webmunk-ext/src/popup/popup.js

```diff
@@ -1,23 +1,99 @@
 import { ENROLL_URL } from '../config';
 
-const continueButton = document.getElementById('continueButton');
-const emailInput = document.getElementById('emailInput');
-const getStartedContainer = document.getElementById('getStartedContainer');
-const studyExtensionContainer = document.getElementById('studyExtensionContainer');
-const copyButton = document.getElementById('copyButton');
-const formattedIdentifier = document.getElementById('formattedIdentifier');
-let fullIdentifier = '';
-
-getStartedContainer.style.display = 'block';
-studyExtensionContainer.style.display = 'none';
-
-document.addEventListener("DOMContentLoaded", () => {
-  displayIdentifier();
-  loadSurveyUrls();
-});
-
-function loadSurveyUrls() {
-  chrome.storage.local.get('surveys', (result) => {
+class EnrollmentManager {
+  constructor() {
+    this.continueButton = document.getElementById('continueButton');
+    this.emailInput = document.getElementById('emailInput');
+    this.getStartedContainer = document.getElementById('getStartedContainer');
+    this.studyExtensionContainer = document.getElementById('studyExtensionContainer');
+    this.copyButton = document.getElementById('copyButton');
+    this.formattedIdentifier = document.getElementById('formattedIdentifier');
+    this.fullIdentifier = '';
+
+    this.init();
+  }
+
+  init() {
+    this.initListeners();
+    this.initView();
+    this.initSurveys();
+  }
+
+  initListeners() {
+    this.continueButton.addEventListener('click', () => this.onContinueButtonClick());
+    this.copyButton.addEventListener('click', () => this.copyIdentifier());
+  }
+
+  async onContinueButtonClick() {
+    const email = this.emailInput.value.trim().toLowerCase();
+
+    if (!email) {
+      alert('E-Mail Required\nPlease enter an e-mail address to continue.');
+      return;
+    }
+
+    this.setButtonState(true, 'Wait...');
+
+    const identifier = await this.getIdentifier(email);
+
+    if (!identifier) {
+      alert('Enrollment hiccup!\nPlease give it another shot a bit later. We appreciate your patience!');
+      this.setButtonState(false, 'Continue');
+      return;
+    }
+
+    await chrome.storage.local.set({ identifier: identifier }, () => {
```

Comment:
You don't need callback here:

await chrome.storage.local.set({ identifier: identifier });
this.showStudyExtensionContainer(identifier);
chrome.runtime.sendMessage({ action: 'cookiesAppMgr.checkPrivacy' });